keps/sig-storage/3756-volume-reconstruction/README.md

```diff
@@ -0,0 +1,1030 @@
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+We have unit tests for the feature disabled or enabled.
+It affects only kubelet startup and we don't change format of data present in
+the OS (mount table, content of `/var/lib/kubelet/pods/`), so we don't have
+automated tests to start kubelet with the feature enabled and then disable it
+or a vice versa.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+If this feature is buggy, kubelet either does not come up at
+all (crashes, hangs) or does not unmount volumes that it should unmount.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+`reconstructed_volumes_total`, `force_cleaned_failed_volumes_total`,
+`orphaned_volumes_cleanup_errors_total`
+
+See Observability in the detail design section. All newly introduced metrics
+will be added both to "old" and "new" VolumeManager, so users can compare
+these metrics with the feature gate enabled and disabled and see if downgrade
+actually helped.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Not yet. This will be a manual test.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+TODO whole chapter before GA.
```

Comment:
phrasing of the comment could use improvement, these question are required for beta.

Reply:
Removed the comment and filed the sections below.